piper/mousemap.py

```diff
@@ -0,0 +1,390 @@
+                    child.widget.unparent()
+                    break
+
+    def do_child_type(self):
+        """Indicates that this container accepts any GTK+ widget."""
+        return Gtk.Widget.get_type()
+
+    def do_forall(self, include_internals, callback, *parameters):
+        """Invokes the given callback on each child, with the given parameters.
+
+        @param include_internals Whether to run on internal children as well,
+                                 as boolean. Ignored, as there are no internal
+                                 children.
+        @param callback The callback to call on each child, as Gtk.Callback
+        @param parameters The parameters to pass to the callback, as object or
+                          None.
+        """
+        if callback is not None:
+            for child in self._children:
+                callback(child.widget, *parameters)
+
+    def do_get_request_mode(self):
+        """Gets whether the container prefers a height-for-width or a
+        width-for-height layout. We don't want to trade width for height or
+        height for width so we return CONSTANT_SIZE."""
+        return Gtk.SizeRequestMode.CONSTANT_SIZE
+
+    def do_get_preferred_height(self):
+        """Calculates the container's initial minimum and natural height. While
+        this call is specific to width-for-height requests (that we requested
+        not to get) we cannot be certain that our wishes are granted and hence
+        we must implement this method as well. We just return the SVG's height
+        plus the border widths.
+        """
+        # TODO: account for children sticking out under or above the SVG, if
+        # they exist. At the moment we cannot reliably do so because the
+        # y-coordinates of the leaders can have any arbitrary value. For now,
+        # we assume that the SVG is high enough to fit all children and do not
+        # worry about setups beyond the default GNOME Adwaita.
+        height = self._handle.props.height + 2 * self.props.border_width
+        return (height, height)
+
+    def do_get_preferred_width(self):
+        """Calculates the container's initial minimum and natural width. While
+        this call is specific to height-for-width requests (that we requested
+        not to get) we cannot be certain that our wishes are granted and hence
+        we must implement this method as well. We return the sum of the SVG's
+        width, the natural child widths (left and right), spacing and border
+        width.
+        """
+        width = 2 * self.props.border_width
+        width_svg = self._handle.props.width
+        width_left = max((child.widget.get_preferred_width()[1] for
+                          child in self._children if child.is_left), default=0)
+        width_right = max((child.widget.get_preferred_width()[1] for
+                          child in self._children if not child.is_left),
+                          default=0)
```

Comment:
nitpick: split the list comprehension up and make this two lines each, it'll be a lot easier to read if you have a
````
left_children = [foo for foo in foo]
width_left = max(left_children)
````